### PR TITLE
MM-16861: Support Guest Authentication via AD/LDAP

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Add UI for LDAPSettings GuestFilter

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-16861

#### Related Pull Requests
- Has server changes (please link here)
https://github.com/mattermost/mattermost-server/pull/12690
https://github.com/mattermost/enterprise/pull/506

### CODE REVIEW

File: components/admin_console/admin_definition.jsx [hunk 1]
```diff
@@ -2389,6 +2389,21 @@ const AdminDefinition = {
...
+                        label_default: 'Guest Filter:',
+                        help_text: t('admin.ldap.guestFilterFilterDesc'),
+                        help_text_default: '(Optional) Enter an AD/LDAP filter to use when searching for guest objects. Only the users selected by the query will be able to access Mattermost as Guests. Guests are prevented from accessing teams or channels upon logging in until they are assigned a team and at least one channel.',
+                        placeholder: t('admin.ldap.guestFilterEx'),
+                        placeholder_default: 'E.g.: "(objectClass=guests)"',
+                        isDisabled: it.both(
+                            it.stateIsFalse('LdapSettings.Enable'),
+                            it.stateIsFalse('LdapSettings.EnableSync'),
+                        ),
+                        isHidden: (config) => it.isnt(it.licensedForFeature('LDAPGroups')) && !config.ServiceSettings.ExperimentalLdapGroupSync,
```
Comment (jespino) [inline components/admin_console/admin_definition.jsx]: I think this feature is not only related to "LDAP Groups" from my perspective it should be valid for any kind of authentication. For example, you have define in your LDAP who is a guest and who is not, so somebody log in for first time in the application, he get a new account with the data that comes from LDAP, and in this case, the role is got too.

File: components/admin_console/admin_definition.jsx [hunk 2]
```diff
@@ -2389,6 +2389,21 @@ const AdminDefinition = {
...
+                        key: 'LdapSettings.GuestFilter',
+                        label: t('admin.ldap.guestFilterTitle'),
+                        label_default: 'Guest Filter:',
+                        help_text: t('admin.ldap.guestFilterFilterDesc'),
+                        help_text_default: '(Optional) Enter an AD/LDAP filter to use when searching for guest objects. Only the users selected by the query will be able to access Mattermost as Guests. Guests are prevented from accessing teams or channels upon logging in until they are assigned a team and at least one channel.',
+                        placeholder: t('admin.ldap.guestFilterEx'),
+                        placeholder_default: 'E.g.: "(objectClass=guests)"',
+                        isDisabled: it.both(
+                            it.stateIsFalse('LdapSettings.Enable'),
+                            it.stateIsFalse('LdapSettings.EnableSync'),
```
Comment (jespino) [inline components/admin_console/admin_definition.jsx]: You don't need to have the sync enable to allow guests to log in as guests, I think is not needed. see the other comment.

File: components/admin_console/admin_definition.jsx (hunk 1 above)
Comment (sbishel) [reply components/admin_console/admin_definition.jsx]: This setting was recently removed from the other settings. I will remove it.

File: components/admin_console/admin_definition.jsx (hunk 2 above)
Comment (sbishel) [reply components/admin_console/admin_definition.jsx]: @jespino These are the same setting as the `UserFilter`.  It is only disabled if BOTH Enable AND EnableSync are false.  So, no, you don't need to have sync enabled to allow guest login.

**Pull request merged**